src/auth.js

```diff
@@ -15,46 +15,70 @@
 
 const {GoogleAuth} = require('google-auth-library');
 const fs = require('fs');
+const util = require('util');
+const exec = util.promisify(require('child_process').exec);
 
 /**
  * Automatically choose the right client credentials based on the environment.
  *
  * @return {!Promise<string>} cred Encrypted access token.
  */
 async function getCreds() {
-  let client;
   try {
-    const auth = new GoogleAuth({
-      scopes: 'https://www.googleapis.com/auth/cloud-platform',
-      projectId: 'unused-project'
-    });
-    client = await auth.getClient();
+    console.log(`Retrieving application default credentials...`)
+    const creds = await getApplicationDefaultCredentials();
+    console.log(`Retrieved application default credentials.`)
+    return creds;
   } catch (err) {
-    throw new Error(
-        'Fail to get credentials. Please run: \n' +
-        '`gcloud auth application-default login` or \n' +
-        '`export GOOGLE_APPLICATION_CREDENTIALS=<path/to/service/account/key>`');
+    console.log(`Failed to retrieve application default credentials: ${err.message}.`);
   }
-
-  let creds;
   try {
-    creds = (await client.getAccessToken()).token;
+    console.log(`Retrieving credentials from gcloud...`)
+    const creds = await getGcloudCredentials();
+    console.log(`Retrived credentials from the current active account logged into gcloud.`)
```

Comment:
Retrieved credentials from the current active account from gcloud.

Reply:
Done.